pkg/liqonet/tunnel/driver.go

```diff
@@ -48,10 +49,12 @@ type Config struct {
 type Driver interface {
 	Init() error
 
-	ConnectToEndpoint(tep *netv1alpha1.TunnelEndpoint) (*netv1alpha1.Connection, error)
+	ConnectToEndpoint(tep *netv1alpha1.TunnelEndpoint, updateStatus conncheck.UpdateFunc) (*netv1alpha1.Connection, error)
 
 	DisconnectFromEndpoint(tep *netv1alpha1.TunnelEndpoint) error
 
+	SetConnectionStatusPerCluster(clusterID string, conn *netv1alpha1.Connection) error
```

Comment:
Seems unnecessary to me, as from the driver you already have access to the connchecker to retrieve the appropriate information.

Reply:
I did it because I used the driver as an interface without casting it to a wireguard object